### ISSUE TITLE
os-mdns-repeater: hammers syslog if one of the repeated networks has no device connected

### ISSUE THREAD

**Issue by triantos** [opnsense/plugins#2869]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

- [ X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md
- [ X] I have searched the existing issues, open and closed, and I'm convinced that mine is new.
- [ X] The title contains the plugin to which this issue belongs

**Describe the bug**
If os-mdns-repeater is enabled to repeat across a few networks and one of them doesn't have equipment connected to the interface, the plugin will bombard system.log with errors that the network is down (I was seeing roughly one message per second). The net result of this is the disk will fill up, and the CPU will be hammered (waiting on I/O) by the syslog-ng process trying to write all the messasges to disk.

**To Reproduce**
1. Install plugin os-mdns-repeater
2. Add one interface with nothing plugged into it (e.g. OPT1)
3. In mdns-repeater setup, have it repeat bonjour packets to the OPT1 interface plus another (e.g. LAN)
4. SSH into the box, and watch the output to /var/log/system/latest.log

What I see is constant messages of the form:
<27>1 2022-02-25T23:59:48-08:00 opnsense.lan mdns-repeater 43304 - [meta sequenceId="55384"] send(): Network is down
<27>1 2022-02-25T23:59:51-08:00 opnsense.lan mdns-repeater 43304 - [meta sequenceId="55385"] send(): Network is down
<27>1 2022-02-25T23:59:51-08:00 opnsense.lan mdns-repeater 43304 - [meta sequenceId="55386"] send(): Network is down
<27>1 2022-02-25T23:59:51-08:00 opnsense.lan mdns-repeater 43304 - [meta sequenceId="55387"] send(): Network is down

**Expected behavior**
Maybe nothing, or maybe complain once that an interface has nothing connected.

**Relevant log files**
See the 4 lines pasted from above

**Additional context**

**Environment**
OPNsense 22.1.1_3-amd64
FreeBSD 13.0-STABLE
OpenSSL 1.1.1m 14 Dec 2021

Running on a Supermicro board with 4 NICs and CPU Intel(R) Pentium(R) CPU N3700 @ 1.60GHz (4 cores, 4 threads)

**Comment by triantos** [comment]
In my case the box became nearly unusable as a result (due to CPU + i/o overload). I shelled into the box and manually killed mdns-repeater to give myself enough cycles to troubleshoot and fix this problem.

**Comment by fichtner** [comment]
@triantos thanks for your analysis. It looks like this should be prevented in the software itself, not the plugin. mdns-repeater, however, is not really maintained so the question is where to report this / carry a fix. Do you agree we should be avoiding the excessive logging in the daemon itself? If yes I would move the ticket to ports.git.

**Comment by triantos** [comment]
Yes my opinion is this is a bug in the actual daemon. If it's really unmaintained then perhaps Opnsense can put a warning in the plugin that "disconnected adapters may cause significant logging" or something like that.

thank you!
